### PR TITLE
Feature :: adding  'add datasource' option  on datasource panel

### LINKED ISSUE #2612: [ Feature ] :: Include "Add datasource" button in query panel
### Is your proposal related to a problem?

Currently we have to add datasource from left sidebar. There is no option to add datasource from the query panel on the go.

### Describe the solution you'd like

Add a button " add datasource " to the right of listed data sources , default sources  will be runjs and REST api so we can place button at last.

This button when clicked can open the same modal for adding datasources used in left sidebar.

Screenshot 2022-03-23 at 1 54 38 PM

### Describe alternatives you've considered
Nothing.

### If the feature is approved, would you be willing to submit a PR?
NO.

### PR DESCRIPTION
**Pull request by stepinfwd**
closes #2612

### CODE REVIEW

File: frontend/src/Editor/QueryManager/DataSourceLister.jsx [hunk 1]
```diff
@@ -34,6 +42,10 @@ function DataSourceLister({ dataSources, staticDataSources, changeDataSource, ha
           </div>
         );
       })}
+      <div className="query-datasource-card" style={computedStyles} onClick={dataSourceModalHandler}>
+        <AddIcon style={{ height: 25, width: 25, marginTop: '-3px' }} />
+        <p style={{ fontWeight: '500' }}> Add datasource</p>
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/QueryManager/DataSourceLister.jsx:47]: Reduce the `fontWeight` to `400` and the icon color to `grey` as it looks different from other sources. 
image

File: frontend/src/Editor/LeftSidebar/index.js [hunk 2]
```diff
@@ -12,31 +12,39 @@ import { LeftSidebarGlobalSettings } from './SidebarGlobalSettings';
...
+    setSelectedComponent,
+    removeComponent,
+    runQuery,
+    toggleAppMaintenance,
+    is_maintenance_on,
+    isSaving,
+    isUnsavedQueriesAvailable,
+  } = props;
   const [showLeaveDialog, setShowLeaveDialog] = useState(false);
+  const [showDataSourceManagerModal, toggleDataSourceManagerModal] = React.useState(false);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/LeftSidebar/index.js:41]: Please replace `React.useState` with `useState` as we're importing it at the top and using it in the file

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@stepinfwd Few minor changes

**Comment by stepinfwd** [comment]
requested changes are  updated @kavinvenkatachalam  @sherfin94

**Review by kavinvenkatachalam — APPROVED** [review]
LGTM

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**